backend/infrahub/core/manager.py

```diff
@@ -1195,29 +1173,127 @@ async def get_many(  # pylint: disable=too-many-branches,too-many-statements
+        peer_ids = grouped_peer_nodes.get_all_peers()
+        # there are no peers to enrich the nodes
+        if not peer_ids:
+            return
+
+        missing_peers: dict[str, Node] = {}
+        if prefetch_relationships:
+            # only query the peers that are not already part of the main list
+            missing_peer_ids = peer_ids - set(nodes_by_id.keys())
+            missing_peers = await cls.get_many(
+                ids=list(missing_peer_ids),
+                branch=branch,
+                at=at,
+                db=db,
+                include_owner=include_owner,
+                include_source=include_source,
+            )
+
+        for node in nodes_by_id.values():
+            await cls._enrich_one_node_with_relationships(
+                db=db,
+                node=node,
+                grouped_peer_nodes=grouped_peer_nodes,
+                nodes_by_id=nodes_by_id | missing_peers,
+                cardinality_one_identifiers_by_kind=cardinality_one_identifiers_by_kind,
+                insert_peer_node=prefetch_relationships,
+            )
+
+    @classmethod
+    async def _enrich_one_node_with_relationships(
+        cls,
+        db: InfrahubDatabase,
+        node: Node,
+        grouped_peer_nodes: GroupedPeerNodes,
+        nodes_by_id: dict[str, Node],
+        cardinality_one_identifiers_by_kind: dict[str, dict[str, RelationshipDirection]] | None,
+        insert_peer_node: bool,
+    ) -> None:
+        if not grouped_peer_nodes.has_node(node_id=node.get_id()):
+            return
+
+        node_schema = node.get_schema()
+        for rel_schema in node_schema.relationships:
+            peer_ids = grouped_peer_nodes.get_peer_ids(
+                node_id=node.get_id(), rel_name=rel_schema.get_identifier(), direction=rel_schema.direction
+            )
+            if not peer_ids:
+                continue
+
+            rel_manager: RelationshipManager = getattr(node, rel_schema.name)
+            if insert_peer_node:
+                rel_peers: list[Node | str] = []
+                for peer_id in peer_ids:
+                    peer = nodes_by_id.get(peer_id)
+                    if peer:
+                        rel_peers.append(peer)
+            # if only getting some relationships, make sure we want THIS relationship for THIS node schema
+            elif cardinality_one_identifiers_by_kind:
+                required_direction = cardinality_one_identifiers_by_kind.get(node_schema.kind, {}).get(
+                    rel_schema.get_identifier()
+                )
+                if required_direction is not rel_schema.direction:
+                    continue
+                rel_peers = list(peer_ids)
+            else:
+                continue
+            if rel_schema.cardinality is RelationshipCardinality.ONE and len(rel_peers) > 1:
+                raise ValueError("At most, one relationship expected")
+
+            rel_manager.has_fetched_relationships = True
```

Comment:
Not directly related to this PR but it would be good for us to build a cleaner interface to update the peer data for a given relationship ... and it would make it easier to read the code